tasks/mysql_secure_installation.yml

```diff
@@ -0,0 +1,63 @@
+---
+
+# supported for ansible ver => 2.0
+#- name: Install python-mysqldb for Ansible
+#  package: pkg=python-mysqldb state=present
+
+
+- name: Install python-mysqldb for Ansible
+  apt: name=python-mysqldb state=present
+  when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu'
+
+- name: Install python-mysqldb for Ansible
+  yum: name=MySQL-python state=present
+  when: ansible_os_family == 'RedHat' or ansible_os_family == 'Oracle Linux'
+
+- name: root password is present
+  mysql_user: name=root host={{ item }} password={{ mysql_root_password }} state=present
+  with_items:
+    - '::1'
+    - '127.0.0.1'
+    - 'localhost'
+
+- name: root has .my.cnf
+  template: src=my.cnf.j2 dest=/root/.my.cnf 
```

Comment:
> If the user running the role is not root but e.g. becomes root with sudo, login will fail

it's not true, at least according my tests

> Also the .my.cnf has to be copied before the "root password is present" task, else the login will fail.

It's not true, it's vice versa. If I'll set up .my.cnf before `root password is present` task - it'll fail if user has empty root password. In current sequence nothing fail in both cases